components/brave_today/browser/publishers_controller.cc

```diff
@@ -51,39 +92,31 @@ PublishersController::~PublishersController() = default;
 
 const mojom::Publisher* PublishersController::GetPublisherForSite(
     const GURL& site_url) const {
-  if (publishers_.empty())
-    return nullptr;
-
   const auto& site_host = site_url.host();
 
   // Can't match a Publisher from an empty host
   if (site_host.empty()) {
     return nullptr;
   }
 
-  for (const auto& kv : publishers_) {
-    const auto& publisher_host = kv.second->site_url.host();
-    // When https://github.com/brave/brave-browser/issues/26092 is fixed, this
-    // hack can be removed.
-    const auto& publisher_host_www = "www." + publisher_host;
-    if (publisher_host == site_host || publisher_host_www == site_host) {
-      return kv.second.get();
-    }
-  }
-
-  return nullptr;
+  return FindMatchPreferringLocale(
+      publishers_, default_locale_,
+      base::BindRepeating(
+          [](const std::string& site_host, const mojom::Publisher& publisher) {
+            return publisher.site_url.host() == site_host;
```

Comment:
totally optional: 

Recently, I noticed a useful function to compare hosts thanks to @nullhook - [net::GetDomainNameAndRegistry](https://source.chromium.org/chromium/chromium/src/+/7c273d0efb14ca53899554bb636edb8e0759eec0:net/base/registry_controlled_domains/registry_controlled_domain.h;l=182)
It looks like we're assuming that hosts in publushers_ are guaranteed to start with "www." but I think it could be fragile. So if we use `GetDomainAndRegistry()` to canoniclize the host parts, we might have more robust code here.

Reply:
The backend has actually been updated, so all publishers will have the right hostname now, so I think this should be okay - `net::GetDomainNameAndRegistry` is sweet though, I've got a few other cases where it's be really handy!